From: Alexander Duyck <alexander.h.duyck@intel.com>
Date: Wed, 23 Dec 2009 13:20:47 +0000
Subject: igb: do not force pcs link when in KX mode
References: FATE#311863 ,bnc#699089
Patch-mainline: v2.6.33-rc3
Git-commit: d68caec645b4b92f6a81985265b024bc94bce41f

We were forcing the PCS link up in error when we are in KX mode.  We should
only be disabling autoneg, not forcing the link up.

Signed-off-by: Alexander Duyck <alexander.h.duyck@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/igb/e1000_82575.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

--- a/drivers/net/igb/e1000_82575.c
+++ b/drivers/net/igb/e1000_82575.c
@@ -1118,9 +1118,7 @@ static s32 igb_setup_serdes_link_82575(s
 			reg |= E1000_PCS_LCTL_FSV_100;
 
 		/* Force speed and force link */
-		reg |= E1000_PCS_LCTL_FSD |
-		       E1000_PCS_LCTL_FORCE_LINK |
-		       E1000_PCS_LCTL_FLV_LINK_UP;
+		reg |= E1000_PCS_LCTL_FSD ; /* Force Speed */
 
 		hw_dbg("Configuring Forced Link; PCS_LCTL = 0x%08X\n", reg);
 	}
